src/hosting/config.ts

```diff
@@ -224,6 +231,41 @@ export function resolveTargets(
   });
 }
 
+/**
+ * Ensures that all configs are of a single modern format
+ */
+export function normalize(configs: HostingMultiple): void {
+  for (const config of configs) {
+    config.rewrites = config.rewrites?.map((rewrite) => {
+      if (!("function" in rewrite)) {
+        return rewrite;
+      }
+      if (typeof rewrite.function === "string") {
+        const modern: HostingRewrites & FunctionsRewrite = {
+          // Note: this copied in a bad "function" and "rewrite" in this splat
+          // we'll overwrite function and delete rewrite.
+          ...rewrite,
+          function: {
+            functionId: rewrite.function,
+            // Do not set pinTag so we can track how often it is used
+          },
+        };
+        delete (modern as unknown as LegacyFunctionsRewrite).region;
+        if ("region" in rewrite && typeof rewrite.region === "string") {
+          modern.function.region = rewrite.region;
+        }
+        // Somebody shoot me... why is it taht these type narrowings compile
+        // but are somehow necessary?
```

Comment:
I'm guessing because `HostingRewrites` is union typed between a bunch of different things, that while technically legal, you have to give it a strong winky face to get the code to behave (that and we don't know the shape of the config's rewrites when we invoke `normalize()`.

Reply:
But `if (typeof rewrite.function === "string")` should indicate that rewrite is a `LegacyFunctionRewrite`, which is where `region` is defined. `rewrite` should have already been typed, but the check against function being a string is only typing that field I think. I may try a different appraoch.